pkgs/development/compilers/oraclejdk/jdk-linux-base.nix

```diff
@@ -67,19 +77,23 @@ let
 
 in
 
+assert architectures ? ${stdenv.hostPlatform.system};
+
 let result = stdenv.mkDerivation rec {
   name =
     if installjdk then "oraclejdk-${productVersion}u${patchVersion}" else "oraclejre-${productVersion}u${patchVersion}";
 
-  src = requireFile {
-    name = {
-      i686-linux    = "jdk-${productVersion}u${patchVersion}-linux-i586.tar.gz";
-      x86_64-linux  = "jdk-${productVersion}u${patchVersion}-linux-x64.tar.gz";
-      armv7l-linux  = "jdk-${productVersion}u${patchVersion}-linux-arm32-vfp-hflt.tar.gz";
-      aarch64-linux = "jdk-${productVersion}u${patchVersion}-linux-arm64-vfp-hflt.tar.gz";
+  src = let
+    platformName = {
+      i686-linux    = "linux-i586";
+      x86_64-linux  = "linux-x64";
+      armv7l-linux  = "linux-arm32-vfp-hflt";
+      aarch64-linux = "linux-arm64-vfp-hflt";
     }.${stdenv.hostPlatform.system};
-    url = downloadUrl;
-    sha256 = sha256.${stdenv.hostPlatform.system};
+  in fetchurl {
+   url = "http://download.oracle.com/otn-pub/java/jdk/${productVersion}u${patchVersion}-b${buildVersion}/${releaseToken}/jdk-${productVersion}u${patchVersion}-${platformName}.tar.gz";
```

Comment:
is there some documentation explaining what oracle means for all these variables, maybe somewhere on their website?

Reply:
Except `releaseToken` this is pretty standard. I could have called the latter one also magic string.

Reply:
If I remember it correctly, the structure of the URLs was different some time ago, so I am against making bets on the current structure to be around in a year or two, and for threating the whole URLs as opaque strings.

Reply:
Example of old-style URLs
https://github.com/bgdevlab/centos-util/blob/54d69f0e98bf98fe6f888c251a12c47f5f24c3d7/scripts/oracle/getOracleJDK.sh

Reply:
And... there is no ```Windows``` and ```Darwin``` versions yet in nixpkgs, but they are not available in ```.tar.gz``` format

Reply:
If the urls breaks that would not be much worse then the current state with `requireFile`. One can still provide a working source in an overlay.

Reply:
Anything is not be worse than ```requireFile``` :)
I just propose
```
 platformName = {
      i686-linux    = "linux-i586";
      x86_64-linux  = "linux-x64";
      armv7l-linux  = "linux-arm32-vfp-hflt";
      aarch64-linux = "linux-arm64-vfp-hflt";
    }
```
to map straigh to the full urls and remove the URL constructor and passing ```releaseToken``` here and there.

Reply:
What use case do you have in mind with this?

Reply:
> What use case do you have in mind with this?

Sorry for being unclear.
I meant that the template
```js
  "http://download.oracle.com/otn-pub/java/jdk/${productVersion}u${patchVersion}-b${buildVersion}/${releaseToken}/jdk-${productVersion}u${patchVersion}-${platformName}.tar.gz";
```
is
1. not "pretty standard" but something introduced about 1-2 years ago, and [probably will be changed in another 1-2 years](https://en.wikipedia.org/wiki/Lindy_effect)
2. even now, it covers only Linux download urls, but not Darwin nor Windows

But what problem does it solve?

Reply:
Ok. Got it. I am not sure if the release token can different between cpu and psu releases. I am also don't know, where I can even get either of these releases. That's why I put this into an external variable.

Reply:
or we can commit it as it is and refactor when the problem arise.
it is a security update so there is no time to think :)